backend/Services/UniquePathGuard.php

```diff
@@ -5,22 +5,18 @@
 namespace Internships\Services;
 
 use Exception;
-use Internships\Models\PathPair;
 
 class UniquePathGuard
 {
     protected array $paths = [];
 
-    public function createPathPair(string $relativePath, string $fileName): PathPair
+    public function verifyIfUnique(string $newPath): void
```

Comment:
The name of this method suggests that it's verifying something, but I can see adding some-other-thing to some array. Maybe you could reconsider choosing another name for that?

Reply:
The array is only used as storage for all used previously paths. It checks if the build is saving/copying some file twice into same path, which shouldn't happen in context of this project.